src/aat/java/uk/gov/hmcts/reform/coh/functional/bdd/steps/AnswerSteps.java

```diff
@@ -1,71 +1,92 @@
 import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpMethod;
 import org.springframework.http.ResponseEntity;
 import org.springframework.test.context.ContextConfiguration;
-import springfox.documentation.spring.web.json.Json;
 import uk.gov.hmcts.reform.coh.controller.answer.AnswerRequest;
 import uk.gov.hmcts.reform.coh.controller.answer.AnswerResponse;
 import uk.gov.hmcts.reform.coh.domain.Answer;
 import uk.gov.hmcts.reform.coh.domain.Question;
-import uk.gov.hmcts.reform.coh.repository.QuestionStateRepository;
+import uk.gov.hmcts.reform.coh.service.AnswerService;
 import uk.gov.hmcts.reform.coh.service.QuestionService;
 
 import java.io.IOException;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
 
 import static org.junit.Assert.assertEquals;
 
 @ContextConfiguration
 @SpringBootTest
-public class AnswerSteps extends BaseSteps{
+public class AnswerSteps extends BaseSteps {
 
     private TestRestTemplate restTemplate = new TestRestTemplate();
 
     private ResponseEntity<String> response;
 
     private String endpoint;
 
-    private Long questionId;
+    private Long currentQuestionId;
 
-    private Long answerId;
+    private Long currentAnswerId;
 
     private Map<String, String> endpoints = new HashMap<String, String>();
 
     private AnswerRequest answerRequest;
 
+    private List<Long> questionIds;
+
+    private List<Long> answerIds;
+
     @Autowired
     private QuestionService questionService;
 
     @Autowired
-    private QuestionStateRepository questionStateRepository;
+    private AnswerService answerService;
 
     @Before
     public void setup() {
         endpoints.put("answer", "/online-hearings/1/questions/question_id/answers");
         endpoints.put("question", "/online-hearings/1/questions");
-        questionId = null;
-        answerId = null;
+
+        currentQuestionId = null;
+        currentAnswerId = null;
+
+        questionIds = new ArrayList<>();
+        answerIds = new ArrayList<>();
+    }
+
+    @After
+    public void cleanup() {
+        /**
+         * For each test run, answers are attached to questions. These need
+         * to be deleted after test completion. Delete in reverse order for
+         * FK constrains
+         */
+        for (Long answerId : answerIds) {
+            answerService.deleteAnswer(new Answer().answerId(answerId));
+        }
+
+        for (Long questionId : questionIds) {
+            questionService.deleteQuestion(new Question().questionId(questionId));
+        }
     }
 
-    @Given("^a valid question$")
     /**
      * Creates a question to be used for testing with an answer
      */
-    public void an_existing_question_with_id() throws IOException {
+    @Given("^a valid question$")
+    public void an_existing_question() throws IOException {
         Question question = new Question();
         question.setSubject("foo");
         question.setQuestionText("question text");
```

Comment:
If the postgres DB has a foreign key constraint on the question 'online_hearing_id', then the test fails(providing there is no online hearing with online_hearing_id = 1. Therefore I think you need to create a online_hearing with id =1 as a prerequisite for the question entity.

Reply:
At this point I don't think there is actually an FK on Online Hearing cause "someone" never put it in and the online hearing/question wasn't ready. That's why the test passes. 
I'm not sure how to handle this. Let's discuss it in the standup